docs/08-upgrading/06-upgrading-from-serverpod_auth.md

```diff
@@ -0,0 +1,253 @@
+
+Replace the `authenticationHandler` in the `Serverpod` instance with the new one like this:
+
+```dart
+import 'package:serverpod_auth_email_server/serverpod_auth_email_server.dart'
+    show AuthSessions;
+
+void run(final List<String> args) async {
+  final pod = Serverpod(
+    args,
+    Protocol(),
+    Endpoints(),
+    authenticationHandler: AuthSessions.authenticationHandler,
+  );
+  
+  …
+}
+```
+
+#### Add backwards compatibility to be able to import legacy sessions and passwords
+
+The migration package stores all legacy sessions and password mapped to the new user IDs in a transitional table. But since we can only fully migrate the passwords once the clients send the clear-text one upon login and migrate sessions on a “per use” basis on demand from the client, we need to add the `serverpod_auth_backwards_compatibility_server` module to the server.
+
+This will automatically expose a new endpoint where updated clients can exchange their legacy session for a new one backed by the `serverpod_auth_session` module.
+
+In order to support importing passwords set in the legacy module into the new `serverpod_auth_email_account` one you have to update your email account endpoint subclass to see whether the password can be imported like this:
+
+```dart
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_auth_backwards_compatibility_server/serverpod_auth_backwards_compatibility_server.dart';
+import 'package:serverpod_auth_email_server/serverpod_auth_email_server.dart'
+    as email_account;
+
+/// Endpoint for email-based authentication which imports the legacy passwords.
+class PasswordImportingEmailAccountEndpoint extends email_account.EmailAccountEndpoint {
+  /// Logs in the user and returns a new session.
+  ///
+  /// In case an expected error occurs, this throws a `EmailAccountLoginException`.
+  @override
+  Future<email_account.AuthSuccess> login(
+    final Session session, {
+    required final String email,
+    required final String password,
+  }) async {
+    await AuthBackwardsCompatibility.importLegacyPasswordIfNeeded(
+      session,
+      email: email,
+      password: password,
+    );
+
+    return super.login(session, email: email, password: password);
+  }
+
+  /// Starts the registration for a new user account with an email-based login associated to it.
+  ///
+  /// Upon successful completion of this method, an email will have been
+  /// sent to [email] with a verification link, which the user must open to complete the registration.
+  @override
+  Future<void> startRegistration(
+    final Session session, {
+    required final String email,
+    required final String password,
+  }) async {
+    await AuthBackwardsCompatibility.importLegacyPasswordIfNeeded(
+      session,
+      email: email,
+      password: password,
+    );
+
+    return super.startRegistration(session, email: email, password: password);
+  }
```

Comment:
```suggestion
```

Remove this here, as the password import at this point would only be useful when `login` is later called, so we can just rely on the import there instead.